### PR TITLE
Forest and foliage overhaul

### PR DESCRIPTION
**Pull request by NetSysFire**
#### Summary
Content "More foliage"

#### Purpose of change

There are many useful plants in the wild, which are important to utilize when you try to survive.

#### Describe the solution

This adds a couple of new trees and plants.

This should make wilderness survival more interesting, diverse and also a tiny bit easier. Easier as in more sources of calories and more importantly: water. If you e.g use the Innawood mod or otherwise start in the wilderness, you might not get clay right away for a clay pot, especially when no clay deposits are near you and you have to dig for clay. A few of the plants, namely the japanese knotwood, can be used for some emergency thirst quenching when nothing else is available. I still need to do work to add other plants I have on my list.

Another important reason is that I absolutely get tired of seeing only 3 types of shrubs.

| Stuff | Explanation |
|---|---|
| Basswood | You can gather some young leaves (which you can also use in a salad) and the flowers are nice, too. |
| Japanese knotweed | Spring vegetable, good when you are just starting out. |
| Hydrangea and Lilacs | There are suprisingly few uses for the flowers, unfortunately. Hydrangea even appears to be toxic due to cyanide. Lilac can be used as a garnish or in honey or syrup. Nothing relevant enough for the game. |
| Wintergreen | A plant which will be insanely useful for a survivor. It is a natural painkiller and can even be harvested in winter. |

I plan on potentially deprecating wild vegetables. That would address #48286

| Planned stuff | Season | Explanation |
|---|---|---|
| Jerusalem artichoke | Fall | Native in Central Nothern America, but also grows in New England. Grows on plains, similar to sunflowers. |
| Wild leek/ramps | Spring | Can be foraged. A wild leafy vegetable. |
| Solomons Seal | Summer | Berries! |
| Groundnuts | Summer | Could be used instead of beans or peas perhaps |
| ~~Indian cucumber~~ | Fall |Also a foragable wild vegetable. Did not make it into the final PR. |
| May apple | Fall | Can be poisonous but due to bugs (#54179) I might not make it actually poisonous |
| Wintergreen | Winter | You can rarely find berries while foraging |

#### Describe alternatives you've considered

None yet.

#### Testing

Playtested extensively. As explained in my last comment, there is a need for additional recipes but I want to be done with this giant PR and will add them in another one.

#### Additional context

Some of my sources:

- https://www.ecolandscaping.org/09/developing-healthy-landscapes/ecological-landscaping-101/edible-and-landscape-worthy-native-plants-of-new-england/
- https://www.ecolandscaping.org/06/designing-ecological-landscapes/native-plants/more-edible-and-landscape-worthy-native-plants-of-new-england/
- https://en.wikipedia.org/wiki/List_of_edible_flowers
- https://en.wikipedia.org/wiki/Lists_of_useful_plants
- https://www.eattheweeds.com/basswood-tree-linden-lime-tree/
- https://www.youtube.com/watch?v=Ys-Mq0YzpVc
- https://www.wildwalks-southwest.co.uk/eating-japanese-knotweed/
- https://plants.ces.ncsu.edu/plants/gaultheria-procumbens/

### CODE REVIEW

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 1]
```diff
@@ -1757,7 +1801,7 @@
     "type": "terrain",
     "id": "t_shrub_hydrangea",
     "name": "hydrangea bush",
-    "description": "A fat, bush of fragrant blue hydrangeas.",
+    "description": "A fat bush of fragrant blue hydrangeas.",
```
Comment (chaohedgehogs) [inline data/json/furniture_and_terrain/terrain-flora.json:1804]:
```suggestion
    "description": "A dense bush of fragrant blue hydrangeas.",
```

File: data/json/furniture_and_terrain/furniture-flora.json [hunk 2]
```diff
@@ -278,6 +278,29 @@
     "harvest_by_season": [ { "seasons": [ "summer", "autumn" ], "id": "mustard_harv" } ],
     "bash": { "str_min": 2, "str_max": 6, "sound": "crunch.", "sound_fail": "whish." }
   },
+  {
+    "type": "terrain",
+    "id": "t_japanese_knotweed",
+    "name": "japanese knotweed",
+    "description": "An invasive species originating from Japan.  The stem looks like it could contain some water.",
```
Comment (chaohedgehogs) [inline data/json/furniture_and_terrain/furniture-flora.json:285]:
```suggestion
    "description": "An invasive species originating from Japan.  The stem looks like it might contain some water.",
```

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 3]
```diff
@@ -263,6 +263,50 @@
       "items": [ { "item": "stick_long", "count": [ 3, 10 ] }, { "item": "splinter", "count": [ 10, 25 ] } ]
     }
   },
+  {
+    "type": "terrain",
+    "id": "t_tree_basswood",
+    "name": "basswood tree",
+    "looks_like": "t_tree",
+    "description": "The American linden.  It has pretty and sweet-smelling flowers in summer and you might be able to eat its young leaves in spring.",
```
Comment (chaohedgehogs) [inline data/json/furniture_and_terrain/terrain-flora.json:271]:
```suggestion
    "description": "The American linden.  It has pretty and sweet-smelling flowers in Summer and you might be able to eat its young leaves in Spring.",
```

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 4]
```diff
@@ -263,6 +263,50 @@
...
+      "ter_set": "t_dirt",
+      "items": [ { "item": "stick_long", "count": [ 3, 10 ] }, { "item": "splinter", "count": [ 10, 25 ] } ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_tree_basswood_harvested",
+    "name": "basswood tree",
+    "looks_like": "t_tree",
+    "description": "The American linden.  It has pretty and sweet-smelling flowers in summer and its young leaves have already been harvested.",
```
Comment (chaohedgehogs) [inline data/json/furniture_and_terrain/terrain-flora.json:294]:
```suggestion
    "description": "The American linden.  It has pretty and sweet-smelling flowers in Summer and its young leaves have already been harvested.",
```

File: data/json/regional_map_settings.json [hunk 5]
```diff
@@ -66,6 +66,7 @@
           "t_tree_birch": 16,
           "t_tree_elm": 16,
           "t_tree_cottonwood": 16,
+          "t_tree_basswood": 16,
```
Comment (Light-Wave) [inline data/json/regional_map_settings.json:69]: Would you mind adding these changes to data/mods/innawood/innawood_map_settings.json as well?
May or may not also want to add them to the following files. Not sure if it fit their respective mod's lore:
data/mods/rural_biome/rural_regional_map_settings.json
data/mods/desert_region/desert_regional_map_settings.json
data/mods/aftershock_exoplanet/region_settings.json
data/mods/classic_zombies/alberta_regional_map_settings.json
data/mods/TropiCataclysm/tropical_regional_map_settings.json
data/mods/No_Hope/regional_map_settings.json

File: data/json/items/comestibles/raw_fruit.json [hunk 6]
```diff
@@ -197,6 +197,27 @@
...
+    "type": "COMESTIBLE",
+    "id": "wintergreen_berry",
+    "name": { "str": "wintergreen berry", "str_pl": "wintergreen berries" },
+    "weight": "3 g",
+    "color": "red",
+    "spoils_in": "5 days",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "healthy": 1,
+    "calories": 20,
```
Comment (Light-Wave) [inline data/json/items/comestibles/raw_fruit.json:210]: Is no quench intentional?

File: data/json/regional_map_settings.json [hunk 7]
```diff
@@ -96,6 +97,7 @@
...
```
Comment (Light-Wave) [inline data/json/regional_map_settings.json:100]: Looks to me like you forgot to make f_wintergreen spawn.

File: data/json/items/comestibles/raw_fruit.json (hunk 6 above)
Comment (NetSysFire) [reply data/json/items/comestibles/raw_fruit.json:210]: Yes. The problem is that I never tasted wintergreen berries in real life and as far as I can tell those berries are unlike blueberries, especially towards the end of winter where they apparently tend to dry out.

File: data/json/regional_map_settings.json (hunk 7 above)
Comment (NetSysFire) [reply data/json/regional_map_settings.json:100]: No, this is still a work in progress. I need to figure out if I make it accessible via foraging only (you randomly get leaves and berries in autumn and winter) or rarely spawn it in the dense forest. Probably both.

File: data/json/regional_map_settings.json (hunk 5 above)
Comment (NetSysFire) [reply data/json/regional_map_settings.json:69]: It seems fitting for No Hope, innawood, rural biome and classic zombies. I would not touch the other mods. I have thought about ignoring mods entirely for now, especially for actively developed mods like innawood because fitting it in there might be tricky in some cases. But I will think about it when I progress further.

Comment (NetSysFire) [reply data/json/regional_map_settings.json:69]: Also I thought about renaming the basswood leaves to more generic young leaves since apparently elm and other trees have edible young leaves, too.

Comment (Light-Wave) [reply data/json/regional_map_settings.json:69]: If this gets accepted to mainline, then, as the developer of innawood, I would much appreciate having the changes in my mod as well. Mirroring your changes from regional_map_settings.json to innawood_map_settings.json should make sure your new flora spawns in the mod too, and I don't think that would cause any issues.

**Comment by NetSysFire** [comment]
So the thing I just pushed needs some further explanation.

My original plan was to make wintergreen spawn in the dense forest more often than in the sparse one, but there was no existing way to do this, so I created sets for the four handled biomes: plains, sparse forest, dense forest and swamps.

I used that opportunity to look into every plant that would be spawned and assigned it to the biomes. So we no longer have sunflowers growing in the forest.

I will probably also tweak the spawning of trees since New England is mostly an Oak-Hickory forest, see https://www.fs.fed.us/nrs/pubs/rb/rb_nrs97.pdf

So far I tested that by giving myself some debug mutations and looking into the forests. It looks nice so far. The forests look rather bleak now (as intended), but there are other things I plan to add to balance that.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/should-fruit-spawns-in-forests-be-nerfed/27538/1

**Comment by NetSysFire** [comment]
I made a decision how to implement the countless plants you can eat greens from (e.g chickweed, violet, several species of nettles etc etc): Generic young leaves. Maybe also shoots and plant marrow.

It appears that harvesting does not work properly yet. I harvested a beech in spring, gathering some young leaves, debug set the time to the last day of summer, waited until it got autumn and you can not harvest again. Is this a known limitation?

https://github.com/CleverRaven/Cataclysm-DDA/pull/54411/commits/3f0eeb9e665cc69d6cc158c09929b75a03c4ec6c#diff-7e45766d1f5c30716bdfcd6a046aed107ae7b77b482ad1a0ef819d677068ca66L15

File: data/json/regional_map_settings.json (hunk 5 above)
Comment (NetSysFire) [reply data/json/regional_map_settings.json:69]: There are already a lot of problems with that. Since I know your mod has the lore of "humans never touched this land", why would there be invasive species from say asia?
When I consider this PR more or less done I will take a look at how much work integrating my content into the different mods will be.

**Comment by NetSysFire** [comment]
Here is a screenshot of the blackberry patch map extra:

image

I am satisfied with it and will work on plant spawning in the forest some more, mostly making sure that the dense forest has more older trees (mostly oaks and hickories) while the sparse forest is considered new growth which will have a different set of trees and shrubs compared to the dense forest.

And I just added twigs and leaves. Both are mostly intended to be burned, but leaves can be used in the pile of leaves recipe. You get both via smashing shrubs and can also find twigs while foraging (because it is in the forest trash group, which has a 10% chance to appear when you forage).

I considered adding them to the filled makeshift shelter and makeshift pillow recipe, but decided against it due to leaves being rather low quality.

Particularly twigs should make finding firewood in the forest without tools easier. I might add twigs to the wattle construction recipes though. Another idea I had was adding a new trap: Crunchy branches. Works just like the bubblewrap trap, you step on them and they break, making noise. A very primitive "trap" which I might also randomly spawn in the forest if there is an easy way to do that (there is one using map extras but I will see about that, mostly worried about performance and cluttering everything with map extras).

File: data/json/regional_map_settings.json [hunk 8]
```diff
@@ -471,6 +471,7 @@
           "mx_spider": 200,
           "mx_nest_wasp": 50,
           "mx_grove": 500,
+          "mx_blackberry_patch": 5000,
```
Comment (anothersimulacrum) [inline data/json/regional_map_settings.json:474]: Weight seems high

File: data/json/regional_map_settings.json [hunk 9]
```diff
@@ -494,7 +495,7 @@
           "mx_science": 20,
           "mx_collegekids": 25,
           "mx_drugdeal": 20,
-          "mx_supplydrop": 10,
+          "mx_supplydrop": 5,
```
Comment (anothersimulacrum) [inline data/json/regional_map_settings.json:498]: Why this change?

**Comment by NetSysFire** [comment]
I just changed the title so it better reflects the changes I have made.

I am currently working on tree distributions in forests. My source for that is https://www.fs.fed.us/nrs/pubs/rb/rb_nrs97.pdf

Here is a table listing the percentages of specific trees being spawned and as you can see I actively tried to have more clean percentages:

| Tree | Before | After |
|---|---|---|
| Oak | 23.52 | 18 |
| Dead Oak | 2.94 | 2<sup>(1)</sup> |
| Young tree | 23.52 | Depends |
| Blackjack Oak | 1.47 | 10 |
| Hickory | 2.94 | 18 |
| Dead Hickory | 2.94 | 2<sup>(1)</sup> |
| Pine | 5.88 | 9 |
| Dead Pine | 2.94 | 1<sup>(1)</sup> |
| Maple | 5.88 | 7 |
| Birch | 2.94 | 7 |
| Beech | 0.36 | 7 |
| Elm | 2.94 | 4 |
| Cottonwood | 2.94 | 4 |
| Cherry | 0.36 | 4 |
| Plum | 0.36 | 2 |
| Others (non-primary) | - | 5 |
| - | - | - |
| Alder | 2.94 | 15 |
| Basswood | 2.94 | 10 |
| Coffee | 0.38 | 1 |
| Walnut | 1.47 | 10 |
| Walnut | 1.47 | 10 |
| Hazelnut | 0.36 | 10 |
| Juniper | 0.36 | 5 |
| Apple | 0.36 | 10 |
| Pear | 0.36 | 1 |
| Peach | 0.36 | 1 |
| Apricot | 0.36 | 2<sup>2</sup> |
| Elderberry | 0.36 | 10 |
| Mulberry | 0.36 | 10 |

1. 10% of the base tree type are dead.
2. Apricot is more tolerating of harsh weather and thus a tiny bit more likely to appear.

My reasoning for altering all this is to better represent the actual distribution of trees in New England. We do not have biomes yet (well, except fields, sparse and dense forests but those are too broad), so it can only be approximate. New England is primarily comprised of Oak-Hickory forests, see page 22 in the pdf. There are also other existing forest types, such as maple-birch-beech forests.
As you can see, those make up about 20% of all the forest land. Both maple, birch and beech trees have a 7% chance each of spawning (21% combined).
Everything that is in the `t_tree_forest_other` group is something I consider non-primary.
Young trees will be more common in the normal/sparse forests.

This just needs some more testing and I will push it out and then work on the map extra spawning to address the criticism in the reviews above.

**Comment by NetSysFire** [comment]
I think the JSON validation is broken here. The tool does not output anything in the tests and I even ran it locally on all files changed by me before pushing and it is not complaining.

**Comment by NetSysFire** [comment]
The test failure seems unrelated.

**Comment by NetSysFire** [comment]
I am aiming to get this done before feature freeze hits on may 1st.

To explain some of the recent changes:

- I have removed peanut bushes from wild spawns because these bushes are definitely not native to New England. They might have been confused with groundnuts (which I just added, which is also why I discovered that peanut bush discrepancy) or hog peanuts.
- I am still adding some more wild plants for more variety. Since not all plants are edible or otherwise useful, I am going to make sure to also add more bushes.
- I figured out how to do the blackberry patch map extra correctly, but that still needs to be implemented. Instead of cranking up the occurances to like 5000, I should make map extras in the forests more common and make like half of them said patches.
- Wild sarsaparilla will be great for Innawoods.

**Comment by Sathra225** [comment]
Are the sarsparilla root and raw artichokes meant to be inedible? If so, not an issue but as the sauteed artichokes use copy_from they are also inedible due to not having a comestible_type: FOOD field.

**Comment by NetSysFire** [comment]
No. Thanks for bringing that to my attention.
